library-app/src/components/Books/BooksList/EditBookForm.tsx

```diff
@@ -0,0 +1,227 @@
+import styles from './ManageBookForm.module.css'
+
+interface EditBookFormProps {
+  book: BookBodyDataGet
+}
+
+const EditBookForm = ({ book }: EditBookFormProps) => {
+  const [authors, setAuthors] = useState<Author[]>([])
+  const [isOpenForm, setIsOpenForm] = useState(false)
+  const [requestCover, setRequestCover] = useState<Blob>(new Blob())
+  const [cover, setCover] = useState('')
+  const [authorForm, setAuthorForm] = useState<AuthorPost>({
+    FirstName: '',
+    LastName: '',
+  })
+  const [formData, setFormData] = useState<BookBodyDataGet>({
+    Id: book.Id,
+    Title: book.Title,
+    Description: book.Description,
+    Isbn: book.Isbn,
+    Quantity: book.Quantity,
+    Cover: book.Cover,
+    PublishDate: book.PublishDate,
+    Authors: book.Authors,
+  })
+
+  const openFormhandler = () => {
+    setIsOpenForm(!isOpenForm)
+  }
+
+  const handleFileChange = ({ currentTarget }: FormEvent<HTMLInputElement>) => {
+    if (currentTarget.files) {
+      const files = currentTarget.files
+      const reader = new FileReader()
+      if (files) {
+        reader.readAsDataURL(files[0])
+        setRequestCover(files[0])
+        reader.onloadend = function () {
+          const base64data = reader.result
+          if (base64data) setCover(base64data as string)
+        }
+      }
+    }
+  }
+
+  const fetchAuthorsData = useCallback(() => {
+    getAuthors().then((response) => {
+      setAuthors(response.data)
+    })
+  }, [])
+
+  useEffect(() => {
+    try {
+      fetchAuthorsData()
+    } catch (error) {
+      if (error) console.error('nema autora')
+    }
+  }, [])
+
+  const editBookHandler = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      const form = new FormData()
+      form.append('Id', book.Id.toString())
+      form.append('Cover', requestCover)
+      form.append('Description', formData.Description)
+      form.append('Isbn', formData.Isbn)
+      form.append('PublishDate', formData.PublishDate)
+      form.append('Quantity', formData.Quantity.toString())
+      form.append('Title', formData.Title)
+      formData.Authors.forEach((author) => form.append('Authors', author.Id.toString()))
+
+      await putBookRequest(form)
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response?.status === 401) {
+        console.error('neautorizovan')
+      }
+      return
+    }
+  }
+
+  const onChangeAuthors = (newAuthors: MultiValue<Author>) => {
+    setFormData((prev) => ({ ...prev, AuthorIds: newAuthors.map((authors) => authors) }))
+  }
+
+  const addAuthorHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    try {
+      const form = new FormData()
+
+      form.append('FirstName', authorForm.FirstName)
+      form.append('LastName', authorForm.LastName)
+      postAuthor(form)
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response?.status === 401) {
```

Comment:
Can you maybe handle 401 error somewhere else? It is not specific for this endpoint